core/store/tx_manager.go

```diff
@@ -370,7 +370,7 @@ func (txm *EthTxManager) BumpGasUntilSafe(hash common.Hash) (*models.TxReceipt,
 	}
 
 	var merr error
-	for attemptIndex := range tx.Attempts {
+	for attemptIndex := len(tx.Attempts) - 1; attemptIndex >= 0; attemptIndex-- {
```

Comment:
Seems like this could go out of date if the attempts ever get ordered somewhere else. The reason for counting down here would not be clear without the context of the story. Can we pull the logic into a method with a descriptive name?